Dart/maths/Euclidean_Algorithm_to_find_GCD.dart

```diff
@@ -0,0 +1,107 @@
+/*
+    Euclidean Algorithm to find GCD in DART
+
+    This algorithm follows the fact that the remainder a' of  a/b is also divisible by the gcd of a and b (i.e. d)
+
+    😊 Lets prove this fact
+    
+    let d be the GCD of a & b 
+    let a>b
+    now    a/b   =>    a   =   a'  +   bq
+    
+    dividing both sides by d, where d is their gcd
+
+    =>  a/d   =   (a'  +   bq)/d
+    =>  a/d   =   a'/d  +   q*(b/d)
+
+    a/d and b/d are in P/Q form where Q=1 since d is their GCD
+    
+    =>  a/d   -   q*(b/d) =   a'/d
+    =>        X           =   a'/d where X is a positive integer
+
+    this also implies that a'/d is also an integer (in p/q form where q=1)
+
+    so it is proved that the remainder a' of  a/b is also divisible by the gcd of a and b (i.e. d)
+    i.e. a,b,d have common gcd (You are going to get it at the end of calculation ---- consider example2)
+
+*/
+
+import 'dart:io';
+
+int gcd(int a, int b) {
+  if (b == 0) return a;
+
+  return gcd(b, a % b);
+  /* 
+    Why return gcd(b, a % b) and not gcd(a % b, b)???
```

Comment:
Example 2 is reffered here.

it supports why in subsequent stack-frame we return gcd(b, a % b) and not gcd(a % b, b)

gcd(100,70)
=gcd(70,30)
=gcd(30,10)
=gcd(10,0)